tests/reference_images.rs

```diff
@@ -95,49 +97,135 @@ fn check_references() {
             let mut path: Vec<_> = path.components().collect();
             (path.pop().unwrap(), path.pop().unwrap())
         };
+
+        let mut filename_parts = filename
+            .as_os_str()
+            .to_str()
+            .unwrap()
+            .split('.')
+            .collect::<Vec<_>>();
+
+        // Ignore the file extension
+        filename_parts.pop().unwrap();
+
+        // The penultimate part of `filename_parts` represents the metadata.
+        let meta_str = filename_parts.pop().unwrap();
+        let mut meta = meta_str.split('_').collect::<Vec<_>>();
+        let ref_crc;
+        let anim_frame: Option<usize>;
+
+        if meta.len() == 1 {
+            // `CRC`
+            ref_crc = meta.pop().unwrap();
+            anim_frame = None;
+        } else if meta.len() == 3 && meta[0] == "anim" {
+            // `anim_FRAME_CRC`
+            ref_crc = meta.pop().unwrap();
+            let frame: usize = meta.pop().unwrap().parse().unwrap();
+            anim_frame = Some(frame.checked_sub(1).expect("frame number cannot be 0"));
+        } else {
+            panic!(
```

Comment:
Rather use `unreachable!` here. `panic` makes it seem like a failing test could trigger this assertion but it should be the structure of the test suite itself. Maybe also include a comment.